### PR TITLE
Updates Neo-3.6.0

### LINKED ISSUE #916: Neo v3.6.0
See Neo v3.6.0 Checklist: https://github.com/neo-project/neo/issues/2844

## Mandatory Changes

- [x] #961
- [x] #941

## New Features

- [x] #917
- [x] #939
- [x] #942
- [x] #945
- [x] #953
- [x] #954
- [x] #955
- [x] #965

See [here](https://github.com/AxLabs/admin-tasks/issues/577) for what's the status on the draft PRs.

#### Related Issues in other Repos:

- [x] https://github.com/AxLabs/neo3-privatenet-docker/issues/24

### LINKED ISSUE #917: [Neo v3.6.0] Add new CryptoLib methods for ZKPs
https://github.com/neo-project/neo/pull/2691

### LINKED ISSUE #939: [Neo v3.6.0] Add FindOptions.Backwards to iterate in reverse order
https://github.com/neo-project/neo/pull/2819

### LINKED ISSUE #941: [Neo v3.6.0] Add lastGasPerVote to NeoAccountState
https://github.com/neo-project/neo/pull/2841/

### LINKED ISSUE #942: [Neo v3.6.0] Add TransactionAttribute `NotValidBefore`
https://github.com/neo-project/neo/pull/2812

### LINKED ISSUE #945: [Neo v3.6.0] Add TransactionAttribute `Conflicts`
https://github.com/neo-project/neo/pull/2818

### LINKED ISSUE #953: [Neo v3.6.0] Add new PRC `FindStorage`
https://github.com/neo-project/neo-modules/pull/805

### LINKED ISSUE #954: [Neo v3.6.0] Add new opcode `ABORTMSG`
https://github.com/neo-project/neo-vm/commit/7e5996844a90b514739f879bc9f873f9a34c9a67

### LINKED ISSUE #955: [Neo v3.6.0] Add new opcode `ASSERTMSG`
https://github.com/neo-project/neo-vm/commit/7e5996844a90b514739f879bc9f873f9a34c9a67

### LINKED ISSUE #961: Compiler uses `PUSH1` instead of `PUSHT` in boolean event parameter
**Summary or problem description**
The neow3j compiler uses the opcode `PUSH1` instead of `PUSHT` in the following event parameter:

```java
private static Event1Arg<Boolean> event;

public static void someMethod() {
    event.fire(true);
}
```

The JVM instructions of `someMethod` contains `ICONST_1`followed by a method call to `Boolean.valueOf()`. Neow3j converts the `ICONST_1` to `PUSH1` and ignores the following usually redundant `valueOf` method call.

The NeoVM version 3.6.0 introduced more restrictive type checks, such that when invoking `someMethod` the NeoVM throws an exception due to a type mismatch.

**Do you have any solution you want to propose?**
The specific case of JVM instructions with a `Boolean.valueOf()` should be exceptionally handled, unlike other `valueOf()` method calls. In the above case, `PUSHT` must be used.

**What part of neow3j does this apply to?**
- Compiler

### LINKED ISSUE #964: More restrictive use of syscall `Notify`
**Description of the bug**
The use of an `Instruction` annotation with the interopService `Notify` on a manually created method might run into problems when that method is called with a event name that does not exist in the manifest of the contract.

**To Reproduce**
For example, if the following method is invoked with the parameter `eventName` being a string value that does not match any name of an event var (that implements `EventInterface`), the NeoVM will exit because there's no such event declared in the manifest.

```java
@Instruction(interopService = InteropService.SYSTEM_RUNTIME_NOTIFY)
public static native void method(String eventName, Object[] state);
````

**Expected behavior**
Suggestion: neow3j should have some documentation on the `Notify` syscall that it should be used carefully in that regard.

**Platform and Software**
Issue arose based on changes introduced with Neo v3.6.0 (affects only unreleased neow3j implementation)

### LINKED ISSUE #965: Add `NeoToken.unclaimedGas()` without block height parameter
**Summary or problem description**
The internal implementation of `NeoToken.unclaimedGas(UInt160 account, uint end)` (see [here](https://github.com/neo-project/neo/blob/4e9314d6b33fb8158e4f2527f889a9cbe7377a73/src/Neo/SmartContract/Native/NeoToken.cs#L337) and [here](https://github.com/neo-project/neo/blob/4e9314d6b33fb8158e4f2527f889a9cbe7377a73/src/Neo/SmartContract/Native/NeoToken.cs#L173-L174C35)) changed, such that only the current height is accepted for the `end` parameter.

**Do you have any solution you want to propose?**
Neow3j should provide a method without the `blockHeight` (i.e., `end`) parameter and fetch the current block height for the dev automatically.

**What part of neow3j does this apply to?**
- SDK

### PR DESCRIPTION
**Pull request by mialbu**
Closes #916 

Closes #917
Closes #939 
Closes #941
Closes #942 
Closes #945 
Closes #953 
Closes #954 
Closes #955 
Closes #964 
Closes #965 

Fixes #961 

Usually, we directly merge new features or fixes into main. However, with major Neo releases, I prefer to collect the development process in a separate branch until the changes and additions for that major release are ready and stable - hence this separate branch.

### CODE REVIEW

File: int-tests/src/test-integration/java/io/neow3j/protocol/Neow3jReadOnlyIntegrationTest.java [hunk 1]
```diff
@@ -851,7 +851,7 @@ public void testListPlugins() throws IOException {
                 .getPlugins();
 
         assertNotNull(plugins);
-        assertThat(plugins, hasSize(10));
+        assertThat(plugins, hasSize(11));
```
Comment (csmuller) [inline int-tests/src/test-integration/java/io/neow3j/protocol/Neow3jReadOnlyIntegrationTest.java:854]: @mialbu, out of curiosity, what's the new plugin that makes this number increment? I am badly informed about the changes in 3.6.0

Comment (mialbu) [reply int-tests/src/test-integration/java/io/neow3j/protocol/Neow3jReadOnlyIntegrationTest.java:854]: > @mialbu, out of curiosity, what's the new plugin that makes this number increment? I am badly informed about the changes in 3.6.0

[`StorageDumper`](https://github.com/neo-project/neo-modules/pull/790)

**Comment by mialbu** [comment]
The codacy analysis fails because of smart contract code for neow3j tests. These can be ignored.

**Comment by csmuller** [comment]
integration test succeed locally, problems with CI/CD will be resolved as soon as we publish the latest neo-privatenet-docker.

**Pull request merged**